libs/sync/files_state.go

```diff
@@ -0,0 +1,103 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/cli/libs/fileset"
+)
+
+// FilesState keeps track of files on the local filesystem and their corresponding
+// entries in WSFS.
+type FilesState struct {
+	// Map of local file names to their last recorded modified time. Files found
+	// to have a newer mtime have their content synced to their remote version.
+	LastModifiedTimes map[string]time.Time `json:"last_modified_times"`
+
+	// Map of local file names to corresponding remote names.
+	// For example: A notebook named "foo.py" locally would be stored as "foo"
+	// in WSFS, and the entry would be: {"foo.py": "foo"}
+	LocalToRemoteNames map[string]string `json:"local_to_remote_names"`
+
+	// Inverse of LocalToRemoteNames. Together they form a 1:1 mapping where all
+	// the remote names and local names are unique.
+	RemoteToLocalNames map[string]string `json:"remote_to_local_names"`
+}
+
+// Convert an array of files on the local file system to a FilesState representation.
+func toFilesState(ctx context.Context, localFiles []fileset.File) (*FilesState, error) {
```

Comment:
`NewSnapshotState` -- and looks like it doesn't need to take a context.

Reply:
done